unit/java_bytecode/java_bytecode_parse_generics/parse_generic_superclasses.cpp

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing classes with generic superclasses with
```

Comment:
The test looks at interfaces too, consider adding it to the description and maybe also changing the name of the file to `parse_generic_bases` (then the question is whether it's not better to merge it with the `parse_derived_generics` which in this light also has an imprecise name).